packages/core/src/client.ts

```diff
@@ -316,16 +316,19 @@ export abstract class Client<O extends ClientOptions = ClientOptions> {
    * @returns A promise that will resolve with `true` if all events are sent before the timeout, or `false` if there are
    * still events in the queue when the timeout is reached.
    */
-  public flush(timeout?: number): PromiseLike<boolean> {
+  // @ts-expect-error - PromiseLike is a subset of Promise
+  public async flush(timeout?: number): PromiseLike<boolean> {
```

Comment:
So for a moment, I thought the `async` signature change would still be breaking for anyone currently extending `Client` and overriding this method (as in, them not having the `async` declaration). But it seems like this is fine (at least with the TS settings in our repo) and TS is smart enough to handle it. Just wanted to leave this thought here in case we can think of a TS setting or a scenario where this might be problematic.